lib/bing-ads-reporting/ad_insight_service.rb

```diff
@@ -0,0 +1,90 @@
+require_relative 'core_service'
+
+module BingAdsReporting
+  class AdInsightService < CoreService
+    WDSL = 'https://reporting.api.bingads.microsoft.com/Api/Advertiser/Reporting/V11/ReportingService.svc?singleWsdl'.freeze
+    FAILED_STATUS = 'Error'.freeze
+    SUCCESS_STATUS = 'Success'.freeze
+
+    private
+
+    def wdsl
+      WDSL
+    end
+
+    def failed_status
+      FAILED_STATUS
+    end
+
+    def success_status
+      SUCCESS_STATUS
+    end
+
+    def report_operation(_)
+      :submit_generate_report
+    end
+
+    def generate_report_message(options)
+      period = options[:period]
+      report_type = options[:report_type]
+
+      { ns('ReportRequest') => {
+          ns('Format') => options[:report_format],
+          ns('Language') => 'English',
+          ns('ReportName') => options[:report_name],
+          ns('ReturnOnlyCompleteData') => 'false',
+
+          ns('Aggregation') => options[:aggregation],
+          ns('Columns') => {
+            ns("#{report_type}ReportColumn") => options[:columns]
+          },
+          ns('Scope') => {
+            ns('AccountIds') => {
+              'arr:long' => @settings[:accountId]
+            }
+          },
+          ns('Time') => {
+            # apparently order is important, and end date has to be before start date, wtf
+            ns('CustomDateRangeEnd') => {
+              ns('Day') => period.to.day,
+              ns('Month') => period.to.month,
+              ns('Year') => period.to.year
+            },
+            ns('CustomDateRangeStart') => {
+              ns('Day') => period.from.day,
+              ns('Month') => period.from.month,
+              ns('Year') => period.from.year
+            }
+            # ns('PredefinedTime') => options[:time]
```

Comment:
Can we remove this if we don't need it anymore?

Reply:
removed